### PR TITLE
Generalize ComponentDiscovery to any android Service.

### PR DESCRIPTION
**Pull request by vkryachko**
This allows users to declare their own component graphs in their own
dedicated Service.

### CODE REVIEW

File: firebase-components/src/main/java/com/google/firebase/components/ComponentDiscovery.java [hunk 1]
```diff
@@ -56,8 +57,29 @@
   private final T context;
   private final RegistrarNameRetriever<T> retriever;
 
+  /**
+   * @deprecated It's unsafe to use when firebase_common is not available. Use {@link
+   *     #forContext(Context, Class)} instead.
+   * @throws IllegalStateException if firebase_common is not available.
+   */
+  @Deprecated
   public static ComponentDiscovery<Context> forContext(Context context) {
```
Comment (sheepmaster) [inline firebase-components/src/main/java/com/google/firebase/components/ComponentDiscovery.java]: When would someone call this deprecated method?

Comment (vkryachko) [reply firebase-components/src/main/java/com/google/firebase/components/ComponentDiscovery.java]: I guess my thinking was that the old firebase-common would call it, although on second thought, old firebase-common will not work with firebase-components anyway as they would have duplicate symbols. Will delete this method then

**Pull request merged**